### PR TITLE
frontend/feat: Estimates screen unification and revamp

### PR DESCRIPTION
**Pull request by aryan-13**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Estimates screen unification and revamp
FIGMA -> [LINK](https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=94705-19366&mode=design&t=ikazFVLwZLcg5HMp-4)
OUTPUT -> [LINK](https://drive.google.com/drive/folders/1ydafW23FrHn2s6H14lNtUt_o3BGgLKEe?usp=share_link)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Resources/Constants.purs [hunk 1]
```diff
@@ -280,11 +280,11 @@ getVehicleCapacity variant = case getMerchant FunctionCall of
           Just ST.TAXI -> getString ECONOMICAL <> " · " <>  "4 " <> getString PEOPLE
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Resources/Constants.purs:280]: @aryan-13 make sure the design changes works with Yatri Sathi as well.

File: Frontend/ui-customer/src/Resources/Constants.purs [hunk 2]
```diff
@@ -280,11 +280,11 @@ getVehicleCapacity variant = case getMerchant FunctionCall of
...
-  YATRI -> case fetchVehicleVariant variant of
-          Just ST.SUV -> "6 " <> (getString SEATS)
-          Just ST.AUTO_RICKSHAW -> "3 " <> (getString SEATS)
-          _ -> "4 " <> (getString SEATS)
-  _ ->    ""
+  _ -> case fetchVehicleVariant variant of
+          Just ST.SUV -> "6" 
+          Just ST.AUTO_RICKSHAW -> "3"
+          _ -> "4" 
+  -- _ ->    ""
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Resources/Constants.purs:287]: remove commented code

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 3]
```diff
@@ -1340,6 +1342,31 @@ sourceToDestinationConfig state = let
     }
   in sourceToDestinationConfig'
 
+chooseVehicleConfig :: ST.HomeScreenState -> ChooseVehicle.Config
+chooseVehicleConfig state = let
+  config = ChooseVehicle.config
+  chooseVehicleConfig' = config
+    { vehicleImage = "ny_ic_auto_quote_list"
+    , isSelected = true
+    , vehicleVariant = state.data.selectedEstimatesObject.vehicleVariant
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs]: put `state.data.selectedEstimatesObject` inside a variable and use that in place of `state.data.selectedEstimatesObject`

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 4]
```diff
@@ -2362,6 +2382,22 @@ eval MapReadyAction state = continueWithCmd state [ do
...
+  void $ pure $ performHapticFeedback unit
+  let event = if autoAssign then "ny_user_pref_autoassigned" else "ny_user_pref_driveroffers"
+  let _ = unsafePerformEffect $ logEvent state.data.logField event
+  void $ pure $ setValueToLocalStore FLOW_WITHOUT_OFFERS (show autoAssign)
+  void $ pure $ setValueToLocalStore TEST_MINIMUM_POLLING_COUNT $ if autoAssign then "4" else "17"
+  void $ pure $ setValueToLocalStore TEST_POLLING_INTERVAL $ if autoAssign then "8000.0" else "1500.0"
+  void $ pure $ setValueToLocalStore TEST_POLLING_COUNT $ if autoAssign then "22" else "117"
+  continue state{props{flowWithoutOffers = (show autoAssign) == "true"}}
+
+eval (ChooseYourRideAction (ChooseYourRideController.OnIconClick autoAssign)) state = do
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:2395]: remove do

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 5]
```diff
@@ -2362,6 +2382,22 @@ eval MapReadyAction state = continueWithCmd state [ do
...
 
+eval (ChooseYourRideAction( ChooseYourRideController.RadioButtonClick autoAssign)) state = do
+  void $ pure $ performHapticFeedback unit
+  let event = if autoAssign then "ny_user_pref_autoassigned" else "ny_user_pref_driveroffers"
+  let _ = unsafePerformEffect $ logEvent state.data.logField event
+  void $ pure $ setValueToLocalStore FLOW_WITHOUT_OFFERS (show autoAssign)
+  void $ pure $ setValueToLocalStore TEST_MINIMUM_POLLING_COUNT $ if autoAssign then "4" else "17"
+  void $ pure $ setValueToLocalStore TEST_POLLING_INTERVAL $ if autoAssign then "8000.0" else "1500.0"
+  void $ pure $ setValueToLocalStore TEST_POLLING_COUNT $ if autoAssign then "22" else "117"
+  continue state{props{flowWithoutOffers = (show autoAssign) == "true"}}
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:2393]: use `toBool` function from MobilityPrelude

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Transformer.purs [hunk 6]
```diff
@@ -324,7 +324,7 @@ getSpecialZoneQuote quote index =
       , index = index
       , id = trim quoteEntity.id
       , capacity = getVehicleCapacity quoteEntity.vehicleVariant
-      , showInfo = (getMerchant FunctionCall) == YATRI
+      , showInfo = (getMerchant FunctionCall) /= YATRISATHI
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Transformer.purs:327]: create a variable in config for this.

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 7]
```diff
@@ -1299,21 +1300,44 @@ rideRequestFlowView push state =
...
+        else
+          emptyTextView state
+
+      isCityInList :: City -> Array City -> Boolean
+      isCityInList city = any (_ == city)
+
+      isStageInList :: Stage -> Array Stage -> Boolean
+      isStageInList stage = any (_ == stage)
+
+primaryButtonRequestRideConfig :: HomeScreenState -> PrimaryButton.Config
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:1329]: move it to ComponentConfig

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Review by kavya-shree-s — DISMISSED** [review]

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**